src/UsgsAstroLsSensorModel.cpp

```diff
@@ -704,112 +666,76 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
-   csm::ImageCoord approxNextPoint = approxPoint;
-   if (approxNextPoint.line + 1 < m_nLines) {
-      ++approxNextPoint.line;
-   }
-   else {
-      --approxNextPoint.line;
-   }
-   double height, aPrec;
-   computeElevation(ground_pt.x, ground_pt.y, ground_pt.z, height, aPrec, desired_precision);
-   csm::EcefCoord approxIntersect = imageToGround(approxPoint, height);
-   csm::EcefCoord approxNextIntersect = imageToGround(approxNextPoint, height);
-   double lineDX = approxNextIntersect.x - approxIntersect.x;
-   double lineDY = approxNextIntersect.y - approxIntersect.y;
-   double lineDZ = approxNextIntersect.z - approxIntersect.z;
-   double approxLineRes = sqrt(lineDX * lineDX + lineDY * lineDY + lineDZ * lineDZ);
-   // Increase the precision by a small amount to ensure the desired precision is met
-   double pixelPrec = desired_precision / approxLineRes * 0.9;
-
-   // Start secant method search on the image lines
-   double sampCtr = m_nSamples / 2.0;
-   double firstTime = getImageTime(csm::ImageCoord(0.0, sampCtr));
-   double lastTime = getImageTime(csm::ImageCoord(m_nLines, sampCtr));
-   // See comment above for why (- 0.5)
-   double firstOffset = computeViewingPixel(firstTime, ground_pt, adj, pixelPrec/2).line - 0.5;
-   double lastOffset = computeViewingPixel(lastTime, ground_pt, adj, pixelPrec/2).line - 0.5;
-   MESSAGE_LOG(m_logger, "groundToImage: Initial firstOffset {}, lastOffset {}", firstOffset, lastOffset)
-
-   double closestLine;
-   csm::ImageCoord detectorCoord;
-
-   // Start secant method search
-   for (int it = 0; it < 30; it++) {
-      double nextTime = ((firstTime * lastOffset) - (lastTime * firstOffset))
-                      / (lastOffset - firstOffset);
-      // Because time across the image is not continuous, find the exposure closest
-      // to the computed nextTime and use that.
-
-      // I.E. if the computed nextTime is 0.3, and the middle exposure times for
-      // lines are 0.07, 0.17, 0.27, 0.37, and 0.47; then use 0.27 because it is
-      // the closest middle exposure time.
-      auto referenceTimeIt = std::upper_bound(m_intTimeStartTimes.begin(),
-                                              m_intTimeStartTimes.end(),
-                                              nextTime);
-      if (referenceTimeIt != m_intTimeStartTimes.begin()) {
-         --referenceTimeIt;
-      }
+   csm::ImageCoord approxPt;
+   computeLinearApproximation(groundPt, approxPt);
+
+   std::vector<double> detectorView;
+   double detectorLine = m_nLines;
+   double detectorSample;
+   double count = 0;
+   double timei;
+
+   while(abs(detectorLine) > desiredPrecision) {
+     if (++count > 15) {
+       if (warnings) {
+         warnings->push_back(
+           csm::Warning(
+             csm::Warning::PRECISION_NOT_MET,
+             "Unable to achieve desired precision within 15 iterations.",
+             "UsgsAstroLsSensorModel::groundToImage()"));
+       }

```

Comment:
I think we can actually remove this warning here because it will happen later on.